core/opendaq/reader/src/stream_reader_impl.cpp

```diff
@@ -230,6 +230,15 @@ ErrCode StreamReaderImpl::markAsInvalid()
 
 void StreamReaderImpl::inferReaderReadType(const DataDescriptorPtr& newDescriptor, std::unique_ptr<Reader>& reader) const
 {
+    if (newDescriptor.isStructDescriptor())
+    {
+        if (newDescriptor.getRawSampleSize() != newDescriptor.getSampleSize())
+            throw InvalidParameterException("Cannot use the reader on implicit or scaled structs");
```

Comment:
Is this the correct exception to throw here?

Reply:
Maybe `InvalidState` would be better or just invalidate the reader

Reply:
Yes, probably invalidating the reader is the way to go. Thanks